src/lib/annotations/doc/DocDrawingThread.js

```diff
@@ -134,23 +139,18 @@ class DocDrawingThread extends DrawingThread {
      * @return {void}
      */
     saveAnnotation(type, text) {
-        this.emit('annotationevent', {
-            type: 'drawcommit'
-        });
         this.reset();
 
         // Only make save request to server if there exist paths to save
-        const { undoCount } = this.pathContainer.getNumberOfItems();
-        if (undoCount === 0) {
+        if (this.pathContainer.isUndoEmpty()) {
```

Comment:
If you're reusing undo logic to determine if we should save I might rename to just isEmpty since you're already accessing the path container.